citation/caching.py

```diff
@@ -1,9 +1,15 @@
 import itertools
 import logging
 
+from citation.models import Publication
 from django.core.cache import cache
 from django.db import connection
 
+from catalog.core.views import generate_network_graph_group_by_tags, generate_network_graph_group_by_sponsors, \
+                               generate_publication_code_platform_data
+
+from catalog.core.util import RelationClassifier
```

Comment:
same here, extract logic from catalog and push it into citation, then refer to it in catalog